examples/ignition-kargs-helper

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+set -euxo pipefail
+
+# Mount /boot. Note that we mount /boot but we don't unmount it because we
+# are run in a systemd unit with MountFlags=slave so it is unmounted for us.
+bootmnt=/mnt/boot_partition
+mkdir -p ${bootmnt}
+bootdev=/dev/disk/by-label/boot
+mount -o rw ${bootdev} ${bootmnt}
+grubcfg="${bootmnt}/grub/grub.cfg"
+
+kernelopts="$(grep kernelopts= <$grubcfg)"
+
+while [[ $# -gt 0 ]]
+do
+key="$1"
+
+case $key in
+--should-exist)
+    arg="$2"
+    # don't repeat the arg
+    if [[ ! " ${kernelopts[*]} " =~ " ${arg} " ]]; then
+        kernelopts="$kernelopts $arg "
+    fi
+    shift 2
+    ;;
+--should-not-exist)
+    arg="$2"
+    kernelopts="$(echo "$kernelopts" | sed "s|$2||g")"
```

Comment:
`console=ttyS0,115200`, `--should-not-exist 0`.

Or, for that matter, `--should-not-exist kernelopts=`.  It might be cleaner to not carry the `kernelopts=` part in the contents of the variable.

Reply:
I've gone ahead and moved the `kernelopts=` out of the variable contents.

As far as the other part, yeah I think that's not great however I'm not sure the best way to approach it without adding a ton of bash complexity. The shell script needs to be able to take something like `--should-not-exist "karg1 karg2"` so doing a simple split of each kernel argument and comparing strings doesn't work. Do you think we should bite the bullet and add bash code to support say splitting the arrays of each and checking if contiguous elements of a split `kernelopts=` array match the array generated by splitting individual `should-exist` & `should-not-exist` entries?

Reply:
We could use the same trick rdcore does and store `$kernelopts` with a leading and trailing space.  Then we can just `s| $2 | |g`, and drop the leading/trailing spaces when writing `$kernelopts` back to the `grub.cfg`.